awscli/clidocs.py

```diff
@@ -186,18 +190,47 @@ def doc_relateditem(self, help_command, related_item, **kwargs):
         )
         doc.write('\n')
 
-    def _document_enums(self, argument, doc):
+    def _document_enums(self, model, doc):
         """Documents top-level parameter enums"""
-        if hasattr(argument, 'argument_model'):
-            model = argument.argument_model
-            if isinstance(model, StringShape):
-                if model.enum:
-                    doc.style.new_paragraph()
-                    doc.write('Possible values:')
-                    doc.style.start_ul()
-                    for enum in model.enum:
-                        doc.style.li('``%s``' % enum)
-                    doc.style.end_ul()
+        if isinstance(model, StringShape):
+            if model.enum:
+                doc.style.new_paragraph()
+                doc.write('Possible values:')
+                doc.style.start_ul()
+                for enum in model.enum:
+                    doc.style.li('``%s``' % enum)
+                doc.style.end_ul()
+
+    def _document_nested_structure(self, model, doc, stack):
+        """Recursively documents parameters in nested structures"""
+        shape_name = getattr(model, 'name', None)
+        if not shape_name or shape_name in stack:
+            return
+        stack.append(shape_name)
+        if isinstance(model, StructureShape):
+            self._output_struct_data(model.members, doc, stack)
+        if isinstance(model, ListShape):
+            self._output_list_data(model, doc, stack)
+        stack.pop()
+
+    def _output_struct_data(self, members, doc, stack):
+        doc.style.start_ul()
+        for name, shape in members.items():
+            doc.style.start_li()
+            doc.write('**%s** (%s) -- ' % (name, shape.type_name))
```

Comment:
So one thing that I noticed is that the format of nested structure is different than the `Output` section. Is there a particular reason we chose one over the other? I think at the very least we should make sure the format is consistent (e.g. either both use the boto3 style of nested parameter documentation or the aws cli output nested style.) Honestly, it may make sense if we just pull out the logic that we use for the output and generalize it for use between the input and output nested structures as it looks like there is not much difference between the two. That way if we decide we do not like the format in general, we can update it once and it will apply to both places (e.g inputs and outputs) where we document nested structures.